### PR TITLE
`Iris`: Send build result to Pyris

### PR DESCRIPTION
**Pull request by SolizerCodes**
#### General
- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).
#### Server
- [x] **Important**: I implemented the changes with a very good performance and prevented too many (unnecessary) database calls.
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/).
- [x] I added pre-authorization annotations according to the [guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/server/#rest-endpoint-best-practices-for-authorization) and checked the course groups for all new REST Calls (security).
- [x] I documented the Java code using JavaDoc style.

### Motivation and Context
Previously, Iris was not sent the actual build result and build log of the student's submission. This PR changes that.

### Description
I added another parameter to the Map of parameters being sent to Pyris

### Steps for Testing
Changing is not really possible.

#### Performance Review
- [x] I confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance 
- [x] I confirm that the server changes (in particular related to database calls) are implemented with a very good performance
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2

### CODE REVIEW

**Review by SabrinaGlatz — DISMISSED** [review]
the code looks good to me

File: src/main/java/de/tum/in/www1/artemis/service/iris/session/IrisChatSessionService.java [hunk 1]
```diff
@@ -128,10 +132,19 @@ public void requestAndHandleResponse(IrisSession session) {
...
             var latestSubmission = participations.get(0).getSubmissions().stream().max(Comparator.comparing(Submission::getSubmissionDate));
-            latestSubmission.ifPresent(submission -> parameters.put("latestSubmission", submission));
+            if (latestSubmission.isPresent()) {
+                var submission = latestSubmission.get();
+                parameters.put("latestSubmission", submission);
+                if (submission.getLatestResult() != null && submission.getLatestResult().getId() != null) {
+                    var programmingSubmission = programmingSubmissionRepository.findByResultId(latestSubmission.get().getLatestResult().getId());
+                    programmingSubmission.ifPresent(result -> parameters.put("buildResult", result));
+                }
+
```
Comment (JohannesStoehr) [inline src/main/java/de/tum/in/www1/artemis/service/iris/session/IrisChatSessionService.java:146]:
```suggestion
```

File: src/main/java/de/tum/in/www1/artemis/service/iris/session/IrisChatSessionService.java [hunk 2]
```diff
@@ -128,10 +132,19 @@ public void requestAndHandleResponse(IrisSession session) {
...
         var participations = studentParticipationRepository.findByExerciseIdAndStudentIdWithEagerResultsAndLegalSubmissions(exercise.getId(), chatSession.getUser().getId());
         if (participations.size() >= 1) {
             var latestSubmission = participations.get(0).getSubmissions().stream().max(Comparator.comparing(Submission::getSubmissionDate));
-            latestSubmission.ifPresent(submission -> parameters.put("latestSubmission", submission));
+            if (latestSubmission.isPresent()) {
+                var submission = latestSubmission.get();
+                parameters.put("latestSubmission", submission);
+                if (submission.getLatestResult() != null && submission.getLatestResult().getId() != null) {
+                    var programmingSubmission = programmingSubmissionRepository.findByResultId(latestSubmission.get().getLatestResult().getId());
+                    programmingSubmission.ifPresent(result -> parameters.put("buildResult", result));
```
Comment (JohannesStoehr) [inline src/main/java/de/tum/in/www1/artemis/service/iris/session/IrisChatSessionService.java:144]: Why do you use the id of the result to retrieve a programming submission only to the use the result of that submission again?
Can't you directly load the programming submission or even the wanted result?

**Review by JohannesStoehr — CHANGES_REQUESTED** [review]
Left two comments.
Additionally: Should you also include the boolean, if it was a build failure?

**Pull request merged**